### PR TITLE
PIM-7342: Implement findOneBy in InMemoryUserRepository

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/back/Acceptance/User/InMemoryUserRepository.php [hunk 1]
```diff
@@ -87,7 +87,14 @@ public function findBy(array $criteria, array $orderBy = null, $limit = null, $o
      */
     public function findOneBy(array $criteria)
     {
-        throw new NotImplementedException(__METHOD__);
+        foreach ($this->users as $user) {
+            $username = $criteria['username'] ?? null;
```
Comment (ahocquard) [inline tests/back/Acceptance/User/InMemoryUserRepository.php]: Put it outside of the `foreach` with nullability check.

I would put `throw new NotImplementedException(__METHOD__)` for any other criteria filter no?

Comment (SamirBoulil) [reply tests/back/Acceptance/User/InMemoryUserRepository.php]: Good idea

File: tests/back/Acceptance/User/InMemoryUserRepository.php [hunk 2]
```diff
@@ -87,7 +88,18 @@ public function findBy(array $criteria, array $orderBy = null, $limit = null, $o
...
     public function findOneBy(array $criteria)
     {
-        throw new NotImplementedException(__METHOD__);
+        $username = $criteria['username'] ?? null;
+        if (null === $username || count($criteria) > 1) {
+            throw new \InvalidArgumentException('This method only supports finding by "username"');
+        }
+
+        foreach ($this->users as $user) {
+            if (null !== $username && $username === $user->getUsername()) {
```
Comment (ahocquard) [inline tests/back/Acceptance/User/InMemoryUserRepository.php]: useless null check ;)

Comment (SamirBoulil) [reply tests/back/Acceptance/User/InMemoryUserRepository.php]: ^^

**Pull request merged**